tests/meson.build

```diff
@@ -167,6 +168,39 @@ endif
 test_override_options = ['b_sanitize=none']
 
 if library_type == 'static'
+	c_compiler = meson.get_compiler('c')
+	searchdirs = run_command(c_compiler.cmd_array(), '-print-search-dirs',
+				check: true).stdout()
+	searchdirs_arr = searchdirs.split('\n')
+	searchline = 'libraries: ='
+	searchpaths = ''
+
+	foreach line : searchdirs_arr
+		if line.startswith(searchline)
+			searchpaths = line.strip(searchline)
+			break
+		endif
+	endforeach
+
+	if searchpaths == ''
+		error('could not find compiler-specific library directory')
+	endif
+
+	searchpaths_arr = searchpaths.split(':')
+	crtpath = ''
+	fs = import('fs')
+	foreach path : searchpaths_arr
+		if fs.exists(path / 'crtbegin.o')
+			crtpath = path
+			break
+		endif
+	endforeach
+
+	if crtpath == ''
+		error('could not find crtbegin.o/crtend.o')
+	endif
+	test_objects += [crtpath / 'crtbegin.o', crtpath / 'crtend.o']
+
```

Comment:
there is test_sources += crt below, maybe remove that?

Reply:
Yes, we still need to keep `crt` for `crt1.o`.